### DISCUSSION TITLE
Fault Code

### DISCUSSION THREAD

**Discussion by beaune33** [PatrikTrestik/homeassistant-solax-http#39] [Ideas]
The modbus documentation lists registers 0x1E and 0x1F for fault code. Would it be possible to read these regsters via http protocol? Does anybody know the appropriate array indices? And are the possible error codes explained somewhere? The documentation does not explain unfortunately.

**Comment by PatrikTrestik** [comment]
In case of FaultCodes it is probably no go.

All work on Http API was done without documentation. I have to reverse engineer values and indexes.
In case of parameters which I can set from SolaX app, it is prety easy.
But for fault codes I'll have to simulate some fault and than check value. Definitely not a good game 👎 

If SolaX provides some documentation for codes (Modbus or API) I'm happy to add it.
